src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -39,417 +28,560 @@
 import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
 import org.parosproxy.paros.core.scanner.Alert;
 import org.parosproxy.paros.core.scanner.Category;
-import org.parosproxy.paros.network.HtmlParameter;
-import org.parosproxy.paros.network.HttpMessage;
-import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.*;
 import org.zaproxy.zap.model.Vulnerabilities;
 import org.zaproxy.zap.model.Vulnerability;
 
+import java.io.*;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
-* a scanner that looks for known insecure HTTP methods enabled for the URL
-* Note that HTTP methods can be enabled for individual URLs, rather than necessarily just at host level
-* It is also possible for methods to be actually be supported, without being documented by the OPTIONS method, so at High Attack Strength, check that as well (regardless of Threshold).  
-* 
-* @author 70pointer
-*
-*/
+ * a scanner that looks for known insecure HTTP methods enabled for the URL Note
+ * that HTTP methods can be enabled for individual URLs, rather than necessarily
+ * just at host level It is also possible for methods to be actually be
+ * supported, without being documented by the OPTIONS method, so at High Attack
+ * Strength, check that as well (regardless of Threshold).
```

Comment:
Formatting changes should be done in a separate PR.
https://github.com/zaproxy/zaproxy/wiki/DevGuidelines#formatting